src/test_trie.py

```diff
@@ -0,0 +1,86 @@
+# _*_ coding: utf-8 _*_
+"""Test the Trie Algorithm."""
+import pytest
+
+
+# use fixtures later
+def test_empty_insert():
+    """Test inserting an empty string."""
+    from trie import Trie
+    t = Trie()
+    t.insert("")
+    assert t.root == {}
+
+
+def test_single_insert():
+    """Test inserting a single character."""
+    from trie import Trie
+    t = Trie()
+    t.insert("a")
+    assert t.root == {"a": {"$": "$"}}
+
+
+def test_single_capitol_insert():
+    """Test inserting a single character."""
+    from trie import Trie
+    t = Trie()
+    t.insert("A")
+    assert t.root == {"a": {"$": "$"}}
+
+
+def test_token_insert():
+    """Test inserting a whole token."""
+    from trie import Trie
+    t = Trie()
+    t.insert("munir")
+    assert t.root == {"m": {"u": {"n": {"i": {"r": {"$": "$"}}}}}}
+
+
+def test_multipe_token_insert():
+    """Test inserting a whole token."""
+    from trie import Trie
+    t = Trie()
+    t.insert("munir")
+    t.insert("moo")
+    assert t.root == {"m": {"o": {"o": {"$": "$"}}, "u": {"n": {"i": {"r": {"$": "$"}}}}}}
```

Comment:
Dictionaries are by nature unordered. By doing it this way, you are relying on luck that this assert will work.